packages/components/src/DataList/DataList.stories.tsx

```diff
@@ -12,43 +21,113 @@ export default {
+      );
+    },
   ],
 } as ComponentMeta<typeof DataList>;
 
-const Template: ComponentStory<typeof DataList> = args => (
-  <DataList {...args} />
-);
+const Template: ComponentStory<typeof DataList> = args => {
+  const {
+    data,
+    /* See useCollectionQuery for example on how to load more */
+    // refresh,
+    // nextPage,
+    // loadingRefresh,
+    // loadingNextPage,
+    loadingInitialContent,
+  } = useCollectionQuery<ListQueryType>({
+    query: LIST_QUERY,
+    queryOptions: {
+      fetchPolicy: "network-only",
+      nextFetchPolicy: "cache-first",
+      client: apolloClient,
+    },
+    getCollectionByPath(items) {
+      return items?.allPeople;
+    },
+  });
 
-export const Basic = Template.bind({});
-const EmptyStateTemplate: ComponentStory<typeof DataList> = args => (
-  <DataList {...args}>
-    <DataList.EmptyState
-      message="This is empty"
-      action={{
-        label: "Do a thing",
-        onClick: () => alert("A thing has been done"),
+  /* For when we start creating a loading state */
+  // const { loading } = getLoadingState(
+  //   loadingInitialContent,
+  //   loadingRefresh,
+  //   loadingNextPage,
+  // );
+
+  const items = data?.allPeople.edges || [];
+  const mappedData = items.map(({ node }) => ({
+    label: node.name,
+    home: node.homeworld.name,
+    tags: [
+      node.gender,
+      node.hairColor?.split(", "),
+      node.skinColor?.split(", "),
+    ],
+    homePopulation: node.homeworld.population?.toLocaleString(),
+    created: new Date(node.created),
+  }));
+
+  return (
+    <DataList
+      {...args}
+      data={(args.data as typeof mappedData) || mappedData}
+      headers={{
+        label: "Name",
+        home: "Home world",
+        tags: "Attributes",
+        homePopulation: "Home world population",
+        created: "Created",
       }}
-    />
-  </DataList>
-);
-export const EmptyState = EmptyStateTemplate.bind({});
+      loading={loadingInitialContent}
+    >
+      <DataList.Layout>
+        {(item: DataListItemType<typeof mappedData>) => (
+          <Grid alignItems="center">
+            <Grid.Cell size={{ xs: 5 }}>
+              <Grid alignItems="center">
+                <Grid.Cell size={{ xs: 6 }}>{item.label}</Grid.Cell>
+                <Grid.Cell size={{ xs: 6 }}>{item.home}</Grid.Cell>
+              </Grid>
+            </Grid.Cell>
+            <Grid.Cell size={{ xs: 2 }}>{item.homePopulation}</Grid.Cell>
+            <Grid.Cell size={{ xs: 3 }}>{item.tags}</Grid.Cell>
+            <Grid.Cell size={{ xs: 2 }}>{item.created}</Grid.Cell>
+          </Grid>
+        )}
+      </DataList.Layout>
+
+      <DataList.EmptyState
+        message="This is empty"
+        action={{
+          label: "Do a thing",
+          onClick: () => alert("A thing has been done"),
+        }}
+      />
+    </DataList>
+  );
+};
+
+export const Basic = Template.bind({});
 
+export const EmptyState = Template.bind({});
 EmptyState.args = {
-  items: [],
+  data: [],
```

Comment:
Minor thing, but because it uses the `Template` loading can't be set to true and stay there because of the `useCollectionQuery` setting it to false to demonstrate the EmptyState not showing while loading. Non-blocking though

Reply:
oh, hmm. lemme check!